python/cuml/ensemble/randomforest.pyx

```diff
@@ -236,17 +252,23 @@ cdef class RandomForest_impl():
         del(y_m)
         return self
 
-    def predict(self, X):
+    def predict(self, X, convert_dtype=False):
+
+        if X.dtype != self.dtype:
```

Comment:
Why not use the input_to_dev_array here?

Reply:
this is CPU numpy array for C++, so we don't want to transfer to GPU

Reply:
oh, right, RF inference is still on CPU, got it!